internal/core/vulndb/scan/scan_controller.go

```diff
@@ -90,14 +91,8 @@ type ScanResponse struct {
 	Flaws        []flaw.FlawDTO `json:"flaws"`
 }
 
-func (s *httpController) Scan(c core.Context) error {
-	bom := new(cdx.BOM)
-	decoder := cdx.NewBOMDecoder(c.Request().Body, cdx.BOMFileFormatJSON)
-	if err := decoder.Decode(bom); err != nil {
-		return err
-	}
-	normalizedBom := normalize.FromCdxBom(bom, true)
-
+func ScanHelpFunction(c core.Context, bom normalize.SBOM, s *httpController) error {
```

Comment:
We should keep the exported functions to a minimum. Since this is a helper function which only gets used in the same package, it it fine to not export it. By defining it as a method on the httpController you do not need to pass it as parameter. This is just a bit more elegant.

```suggestion
func (s *httpController) scanHelpFunction(c core.Context, bom normalize.SBOM) error {
```